cpp/velox/shuffle/VeloxShuffleWriter.cc

```diff
@@ -594,8 +592,8 @@ arrow::Status VeloxShuffleWriter::stop() {
   setSplitBufferSize(pool_->bytes_allocated());
   EVAL_START("write", options_.thread_id)
   RETURN_NOT_OK(partitionWriter_->stop());
-  if (options_.ipc_memory_pool != options_.memory_pool) {
-    options_.ipc_memory_pool.reset();
+  if (options_.ipc_memory_pool != nullptr && options_.ipc_memory_pool != options_.memory_pool) {
+    delete options_.ipc_memory_pool;
   }
```

Comment:
Why not just leave a smart pointer for options_.ipc_memory_pool since it's just test code? `delete` still looks risky here.

Besides, in previous code why the if condition was needed?

```cpp
if (options_.ipc_memory_pool != options_.memory_pool) {
  options_.ipc_memory_pool.reset();
}
```

Why can't we use
```cpp
options_.ipc_memory_pool.reset();
```
instead? @marin-ma

Reply:
> Why not just leave a smart pointer for options_.ipc_memory_pool since it's just test code?

Two solutions here:
1. keep options_.ipc_memory_pool but comments its usage at `VeloxShuffleWriter#initIpcWriteOptions`, since these codes are not used. 
2. current implementation.

The key is we need set ipc_memory_pool with memory_pool to enable this unused in-test feature, keep these two pool with same type is better.